src/routes/relay/pipes/sponsored-call-dto.validator.pipe.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  PipeTransform,
+  Injectable,
+  HttpException,
+  HttpStatus,
+  Inject,
+} from '@nestjs/common';
+
+import {
+  ISafeInfoService,
+  SafeInfoService,
+} from '../../../datasources/safe-info/safe-info.service.interface';
+import { SponsoredCallSchema } from '../entities/schema/sponsored-call.schema';
+import { SponsoredCallDto } from '../entities/sponsored-call.entity';
+
+@Injectable()
+export class SponsoredCallDtoValidatorPipe implements PipeTransform {
+  static schema = SponsoredCallSchema;
```

Comment:
I don't know Zod that well but I'd say this is safe as long:
- The schema outlives the pipe lifecycle
- There's no internal Zod state kept in the `SponsoredCallSchema` – if there is this might cause issues given that we are sharing the same instance across multiple _potential_ `SponsoredCallDtoValidatorPipe`

Reply:
Very valid points. I've converted this to `private readonly`.